src/pyclaw/fileio/ascii.py

```diff
@@ -329,39 +329,51 @@ def read_array(f, state, num_var):
     q = np.zeros(q_shape)
 
 
-    if patch.num_dim == 1:
-        for i in range(patch.dimensions[0].num_cells):
-            l = []
-            while len(l)<num_var:
-                line = f.readline()
-                l = l + line.split()
-            for m in range(num_var):
-                q[m,i] = float(l[m])
-    elif patch.num_dim == 2:
-        for j in range(patch.dimensions[1].num_cells):
+    try:
+        if patch.num_dim == 1:
             for i in range(patch.dimensions[0].num_cells):
                 l = []
                 while len(l)<num_var:
                     line = f.readline()
+                    if line=='':
+                        raise IOError('Unexpected EOF in %s' % f.name)
                     l = l + line.split()
                 for m in range(num_var):
-                    q[m,i,j] = float(l[m])
-            blank = f.readline()
-    elif patch.num_dim == 3:
-        for k in range(patch.dimensions[2].num_cells):
+                    q[m,i] = float(l[m])
+        elif patch.num_dim == 2:
             for j in range(patch.dimensions[1].num_cells):
                 for i in range(patch.dimensions[0].num_cells):
-                    l=[]
-                    while len(l) < num_var:
+                    l = []
+                    while len(l)<num_var:
```

Comment:
The reformatting here is confusing -- on one line, we've added spaces around a "=" and on the next we've removed it around ">".  Not a big deal but PEP8 says to have spaces in both cases.

Reply:
Thanks I fixed this and a few other places.